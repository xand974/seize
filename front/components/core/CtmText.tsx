import React from "react";
import {
  NativeSyntheticEvent,
  NativeTouchEvent,
  StyleSheet,
  Text,
  View,
} from "react-native";
import tw from "twrnc";
import font from "styles/_font";

interface CtmTextProp {
  style?: string;
  children: React.ReactNode | string;
  type:
    | "MontserratBold"
    | "Asar"
    | "AlexBrush"
    | "MontserratThin"
    | "MontserratLight"
    | "MontserratExtraLight"
    | "MontserratLight"
    | "MontserratRegular"
    | "MontserratMedium"
    | "MontserratSemiBold"
    | "MontserratBold"
    | "MontserratBlack";
  onPress?: (e?: NativeSyntheticEvent<NativeTouchEvent>) => void;
}
export default function CtmText({
  style,
  children,
  type,
  onPress,
}: CtmTextProp) {
  return (
    <Text
      style={tw.style(styles[type], `text-white ${style ? style : ""}`)}
      onPress={onPress}
    >
      {children}
    </Text>
  );
}

const styles = StyleSheet.create({
  MontserratBold: {
    fontFamily: font.MontserratBold.fontFamily,
  },
  Asar: {
    fontFamily: font.Asar.fontFamily,
  },
  AlexBrush: {
    fontFamily: font.AlexBrush.fontFamily,
  },
  MontserratThin: {
    fontFamily: font.MontserratThin.fontFamily,
  },
  MontserratExtraLight: {
    fontFamily: font.MontserratExtraLight.fontFamily,
  },
  MontserratLight: {
    fontFamily: font.MontserratLight.fontFamily,
  },
  MontserratRegular: {
    fontFamily: font.MontserratRegular.fontFamily,
  },
  MontserratMedium: {
    fontFamily: font.MontserratMedium.fontFamily,
  },
  MontserratSemiBold: {
    fontFamily: font.MontserratSemiBold.fontFamily,
  },
  MontserratExtraBold: {
    fontFamily: font.MontserratExtraBold.fontFamily,
  },
  MontserratBlack: {
    fontFamily: font.MontserratBlack.fontFamily,
  },
});
