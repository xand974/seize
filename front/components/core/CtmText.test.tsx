import CtmText from "./CtmText";
import renderer from "react-test-renderer";

describe("render custom text component", () => {
  it("should render the CtmText Component", () => {
    const component = renderer.create(
      <CtmText type="MontserratBold">TEST</CtmText>
    );
    const componentJSON = component.toJSON() as any;
    expect(componentJSON).toMatchSnapshot();
  });
});
