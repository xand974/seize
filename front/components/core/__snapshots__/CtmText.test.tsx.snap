// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render custom text component should render the CtmText Component 1`] = `
<Text
  style={
    {
      "color": "#fff",
      "fontFamily": "MontserratBold",
    }
  }
>
  TEST
</Text>
`;
